drain.go

```diff
@@ -95,6 +96,17 @@ func (s *LumbermillServer) serveDrain(w http.ResponseWriter, r *http.Request) {
 
 		destination := s.hashRing.Get(id)
 
+		// Lock, or don't do any work, but don't block.
+		// This, essentially, samples the incoming tokens for the purposes of health checking
+		// live tokens. Rather than use a random number generator, or a global counter, we
+		// let the scheduler do the sampling for us.
+		if atomic.CompareAndSwapInt32(s.tokenLock, 0, 1) {
```

Comment:
Maybe move this whole thing to a function / method and call it only when handling a dyno log-runtime-metrics message (`case bytes.Contains(msg, dynoMemMsgSentinel)`)